### PR TITLE
Update to the latest dev-support defaults

### PR DESCRIPTION
**Pull request by elia**
## Summary

Just a few cleanups from running `bundle exec solidus extension .`.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [ ] I have added automated tests to cover my changes.
- [ ] I have attached screenshots to demo visual changes.
- [ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- [ ] I have updated the README to account for my changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -384,6 +380,24 @@ $ bin/rails server
...
+
+Before and after releases the changelog should be updated to reflect the up-to-date status of
+the project:
+
+```shell
+bin/rake changelog
+git add CHANGELOG.md
+git commit -m "Update the changelog"
+```
+
```
Comment (waiting-for-dev) [inline README.md:396]: Do we need that? That's also part of the release instructions linked below.

File: solidus_paypal_braintree.gemspec [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
-  spec.required_ruby_version = '>= 2.5'
+  # Specify which files should be added to the gem when it is released.
+  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
+  files = Dir.chdir(__dir__) { `git ls-files -z`.split("\x0") }
 
-  spec.files = Dir.chdir(File.expand_path(__dir__)) do
-    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
-  end
+  spec.files = files.grep_v(%r{^(test|spec|features)/})
+  spec.test_files = files.grep(%r{^(test|spec|features)/})
```
Comment (waiting-for-dev) [inline solidus_paypal_braintree.gemspec:27]: What about removing `test_files`? See https://github.com/rubocop/rubocop/issues/10675

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @elia. I left some comments. Feel free to merge once you review and decide.

**Review by gsmendoza — APPROVED** [review]
I'm good with merging this, though we still need to address @waiting-for-dev 's feedback. We'll also need to make the corresponding updates to solidus_dev_support.

**Pull request merged**